Source/WebCore/rendering/svg/SVGRenderSupport.cpp

```diff
@@ -650,6 +652,10 @@ FloatRect SVGRenderSupport::calculateApproximateStrokeBoundingBox(const RenderEl
     ASSERT(is<RenderSVGShape>(renderer));
     const auto& shape = downcast<RenderSVGShape>(renderer);
     return shape.adjustStrokeBoundingBoxForZeroLengthLinecaps(RepaintRectCalculation::Fast, calculate(shape));
```

Comment:
It seems that you should just guard these lines with `#if ENABLE(LAYER_BASED_SVG_ENGINE)` instead of recopying code into an `#else`—this would match the [earlier case on line 481](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/rendering/svg/SVGRenderSupport.cpp#L481-L484).

Reply:
Right, I'd like to use the coding style but there is one unclear part there. 

[Source/WebCore/rendering/svg/SVGRenderSupport.cpp](https://github.com/WebKit/WebKit/commit/f7f897982b66445d96622a5b67145280e2ba2d81#diff-542b0605fe303b01bd02c8cd36c5e92cb3c5dee6ad232704ecb20bca9ea4f94fR645-R648
)
>     if (is<LegacyRenderSVGShape>(renderer)) {
>         const auto& shape = downcast<LegacyRenderSVGShape>(renderer);
>         return shape.adjustStrokeBoundingBoxForMarkersAndZeroLengthLinecaps(RepaintRectCalculation::Fast, calculate(shape));
>     }
> 

I think this part without "is&lt;LegacyRenderSVGShape&gt;(renderer)) would be a default case when LAYER_BASED_SVG_ENGINE is disabled. Following your suggestion, the part will be

```
#if !ENABLE(LAYER_BASED_SVG_ENGINE)
    if (is<LegacyRenderSVGShape>(renderer)) {
#endif
        const auto& shape = downcast<LegacyRenderSVGShape>(renderer);
        return shape.adjustStrokeBoundingBoxForMarkersAndZeroLengthLinecaps(RepaintRectCalculation::Fast, calculate(shape));
#if !ENABLE(LAYER_BASED_SVG_ENGINE)
    }
#endif

#if ENABLE(LAYER_BASED_SVG_ENGINE)
    ASSERT(is<RenderSVGShape>(renderer));
    const auto& shape = downcast<RenderSVGShape>(renderer);
    return shape.adjustStrokeBoundingBoxForZeroLengthLinecaps(RepaintRectCalculation::Fast, calculate(shape));
#endif
```

So it seems to make the part difficult to read, recopying the code is applied.  But I'm not sure whether my thinking is correct. If it's wrong, this part must be rewritten with a right design. I'd like to know how the part can be fixed through this review.

Reply:
I was suggesting that you follow the earlier code exactly, which means letting the `if` stay, even though it's unnecessary. This policy makes the most sense since it is a build fix.

Reply:
Thanks, I understand, will fix.

Reply:
Ah, sorry, I misunderstood. If all paths need to return a value then your original patch was best. Sorry for the confusion.

Reply:
Don't mind. I could check this PR from wide variety of points. I appreciate your comments.